src/daemon/ca/events.rs

```diff
@@ -71,19 +74,195 @@ impl fmt::Display for IniDet {
 /// Describes an update to the set of ROAs under a ResourceClass.
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct RoaUpdates {
-    #[serde(skip_serializing_if = "HashMap::is_empty", default = "HashMap::new")]
+    #[serde(
+        skip_serializing_if = "HashMap::is_empty",
+        default = "HashMap::new",
+        with = "updated_map"
+    )]
     updated: HashMap<RouteAuthorization, RoaInfo>,
 
-    #[serde(skip_serializing_if = "HashMap::is_empty", default = "HashMap::new")]
+    #[serde(
+        skip_serializing_if = "HashMap::is_empty",
+        default = "HashMap::new",
+        with = "removed_map"
+    )]
     removed: HashMap<RouteAuthorization, RevokedObject>,
 
-    #[serde(skip_serializing_if = "HashMap::is_empty", default = "HashMap::new")]
+    #[serde(
+        skip_serializing_if = "HashMap::is_empty",
+        default = "HashMap::new",
+        with = "aggregate_updated_map"
+    )]
     aggregate_updated: HashMap<RoaAggregateKey, AggregateRoaInfo>,
 
-    #[serde(skip_serializing_if = "HashMap::is_empty", default = "HashMap::new")]
+    #[serde(
+        skip_serializing_if = "HashMap::is_empty",
+        default = "HashMap::new",
+        with = "aggregate_removed_map"
+    )]
     aggregate_removed: HashMap<RoaAggregateKey, RevokedObject>,
 }
 
+mod updated_map {
+    use super::*;
+
+    use serde::de::{Deserialize, Deserializer};
+    use serde::ser::Serializer;
+
+    #[derive(Debug, Deserialize)]
+    struct Item {
+        auth: RouteAuthorization,
+        roa: RoaInfo,
+    }
+
+    #[derive(Debug, Serialize)]
+    struct ItemRef<'a> {
+        auth: &'a RouteAuthorization,
+        roa: &'a RoaInfo,
+    }
+
+    pub fn serialize<S>(map: &HashMap<RouteAuthorization, RoaInfo>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut sorted_vec: Vec<ItemRef> = map.iter().map(|(auth, roa)| ItemRef { auth, roa }).collect();
+        sorted_vec.sort_by_key(|el| el.auth);
```

Comment:
Given that `sort_by_key()` docs say "When applicable, unstable sorting is preferred because it is generally faster than stable sorting and it doesn't allocate auxiliary memory. See sort_unstable_by_key" is `sort_unstable_by_key()` not used in order to have a predictable testable output order? Or could it be used? Or is there some other reason?